[PyTorch Edge][tracing-based] tracing-based build libtorch_lite

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64406
* #64152
* #64087
* #63421

Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

Commit:
[PyTorch Edge][tracing-based] tracing-based build libtorch_lite



Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

malfet commented on caffe2/CMakeLists.txt:
```diff
@@ -1019,14 +1019,24 @@ endif()
+    --output_dir "${CMAKE_BINARY_DIR}/aten/src/ATen"
     WORKING_DIRECTORY "${TORCH_ROOT}")
 
+  # add_custom_command(
+  #   OUTPUT ${CMAKE_BINARY_DIR}/aten/src/ATen/selected_mobile_ops.h
+  #   COMMAND
+  #   "${PYTHON_EXECUTABLE}" -m tools.lite_interpreter.gen_selected_mobile_ops_header
+  #   --yaml_file_path "${SELECTED_OP_LIST}"
+  #   --output_file_path "${CMAKE_BINARY_DIR}/aten/src/ATen"
+  #   WORKING_DIRECTORY "${TORCH_ROOT}")
+
```
Please remove this one
```suggestion

```

malfet commented on tools/lite_interpreter/gen_oplist.py:
```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+import argparse
+import json
```
Please fix flake8 errors

malfet commented on tools/lite_interpreter/gen_oplist.py:
```diff
@@ -0,0 +1,199 @@
+
+import yaml
+from tools.code_analyzer.gen_op_registration_allowlist import canonical_name
+from tools.codegen.selective_build.selector import *
```
Please don't use `import *`

malfet commented on tools/lite_interpreter/gen_oplist.py:
```diff
@@ -0,0 +1,199 @@
+        out_file.write(source.encode("utf-8"))
+
+
+def main(argv):
```
Nit
```suggestion
def main(argv) -> None:
```

malfet commented on tools/lite_interpreter/gen_oplist.py:
```diff
@@ -0,0 +1,199 @@
+    )
+    options = parser.parse_args()
+
+    # assert options.model_file_list_path[0] == "@"
+    # model_file_list_path = options.model_file_list_path[1:]
+
+    # model_dicts = []
+    # with open(model_file_list_path) as model_list_file:
+    #     model_file_names = model_list_file.read().split()
+    #     for model_file_name in model_file_names:
+    #         with open(model_file_name, "rb") as model_file:
+    #             model_dict = yaml.safe_load(model_file)
+    #             model_dicts.append(model_dict)
```
Please remove commented out code from final version of PR
```suggestion
```

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

malfet commented on tools/lite_interpreter/gen_oplist.py:
```diff
@@ -0,0 +1,199 @@
+
+
+if __name__ == "__main__":
+    sys.exit(main(sys.argv))
```
I haven't noticed a single return statement in implementation of `main` routine
```suggestion
    main(sys.argv)
```

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]

Commit:
Update on "[PyTorch Edge][tracing-based] tracing-based build libtorch_lite"


Differential Revision: [D30713988](https://our.internmc.facebook.com/intern/diff/D30713988/)

[ghstack-poisoned]
